dragonflybsd/esrt-api.md

```diff
@@ -0,0 +1,92 @@
+# Instructions for getting user-space ESRT API on DragonFlyBSD
+
+Read [README.md](./README.md) first for general information.
+
+The plan is:
+ * obtain `ioctl-efitool` branch of https://github.com/3mdeb/DragonFlyBSD/tree/esrt
+ fork of DragonFlyBSD source tree
+ * compile kernel
+ * compile EFI bootloader
+ * install both of them
+ * compile and install tool that uses ESRT API
+ * reboot and check that it works
+
+## Preparation
+
+As a root on installed DragonFlyBSD:
+
+```
+# cd /usr
+# make src-create-shallow
+# make src-update
+```
+
+Sources initialization may take a while. After that, add 3mdeb repository:
+
+```
+# git remote add 3mdeb https://github.com/3mdeb/DragonFlyBSD.git
+# git fetch 3mdeb
+# git checkout 3mdeb/ioctl-efitool
+```
+
+## Kernel compilation and installation
+
+```
+# cd /usr/src/sys/config
+# mkdir /root/kernels
+# cp X86_64_GENERIC /root/kernels/MYKERNEL
+# ln -s /root/kernels/MYKERNEL
+
+# make -j4 nativekernel KERNCONF=MYKERNEL
+# make installkernel KERNCONF=MYKERNEL
+```
+
+## EFI bootloader compilation and installation
+
+```
+# cd /usr/src/stand/boot/efi/libefi
+# make
+
+# cd /usr/src/stand/boot/efi/loader
+# make
+# make install
+# reboot
+```
+
+## Prepare user-space tool
+
+```
+# cd /usr/src/usr.sbin/efitable
+# make
+```
+
+## Verification
+
+Kernel and bootloader from 3mdeb sources should be installed which was described
+in [Building custom kernel](#Building-custom-kernel) section. To verify ESRT
+tables functionality you need to load EFI runtime module:
+
+```
+# kldload efirt
```

Comment:
@kotylamichal just loading the module should not print the esrt table, we should remove that from code

Reply:
@TomaszAIR removed in last commit: https://github.com/3mdeb/DragonFlyBSD/pull/6/commits/571b1c18ff8926f95b7302de4e6af14d336f2420

Reply:
Yes, table is not printed now